lib/Fmt_ast.ml

```diff
@@ -1454,15 +1458,22 @@ and fmt_args_grouped ?epi:(global_epi = noop) c ctx args =
       | Pexp_fun _ | Pexp_function _ -> Some false
       | _ -> None
     in
+    let break_expr =
+      match ast.pexp_desc with
+      | (Pexp_match _ | Pexp_apply _) when not last ->
+          let indent = match lbl with Nolabel -> -1 | _ -> -3 in
+          (* Some (break 1000 indent) *)
+          Some (fits_breaks "" ~hint:(1000, indent) "")
+      | _ -> None
+    in
     let epi =
-      match (lbl, last) with
-      | _, true -> None
-      | Nolabel, _ -> Some (fits_breaks "" ~hint:(1000, -1) "")
-      | _ -> Some (fits_breaks "" ~hint:(1000, -3) "")
+      match ast.pexp_desc with
+      | Pexp_constant _ when not last -> fits_breaks "" ~hint:(1000, -2) ""
+      | _ -> noop
```

Comment:
I found out that the first rule was useless. Only this one remains.

The `break` code below could probably handle un-grouping `Pexp_constant` better. This introduces hard to control indentation.

Reply:
This rule is actually more complex than that: it doesn't apply to every kinds of constants, when `break-string-literals=never` and in other cases that I couldn't find.

The regression are down to +126 -80 and are easily explained: Always break after multi-line string argument.

Reply:
The regressions are due to hard to control implicit rules. I suggest making these explicit in https://github.com/ocaml-ppx/ocamlformat/pull/2448 (by introducing more regressions but on purpose)

Reply:
This change is merged in https://github.com/ocaml-ppx/ocamlformat/pull/2453 and this PR has no more regressions!